### PR TITLE
fix!: remove deprecated platforms

### PR DESCRIPTION
**Pull request by timbru31**
### Platforms affected

- Amazon Fire OS
- Blackberry 10
- Windows Phone
- Firefox OS
- Windows 8

### Motivation and Context

Maintain only active platforms.

### Description

Removes deprecated platforms.

### Checklist

- [ ] I've run the tests to see all new and existing tests pass
- [ ] I added automated test coverage as appropriate for this change
- [ ] Commit is prefixed with `(platform)` if this change only applies to one platform (e.g. `(android)`)
- [ ] If this Pull Request resolves an issue, I linked to the issue in the text above (and used the correct [keyword to close issues using keywords](https://help.github.com/articles/closing-issues-using-keywords/))
- [ ] I've updated the documentation if necessary

### CODE REVIEW

File: tests/tests.js [hunk 1]
```diff
@@ -54,8 +54,6 @@ exports.defineAutoTests = function () {
 
     // flags
     var isWindows = cordova.platformId === "windows8" || cordova.platformId === "windows";
-    var isWindowsPhone81 = isWindows && WinJS.Utilities.isPhone;
-    var isWP8 = cordova.platformId === "windowsphone";
```
Comment (erisu) [inline tests/tests.js:58]: Can you confirm this?

```
var isWindows = cordova.platformId === "windows8" || cordova.platformId === "windows";
```
I suspect should become:

```
var isWindows = cordova.platformId === "windows";
```

But then I think the Windows platform can be on a phone? so I believe
```
var isWindowsPhone81 = isWindows && WinJS.Utilities.isPhone;
```
should become
```
var isWindowsPhone = isWindows && WinJS.Utilities.isPhone;
```

File: tests/tests.js [hunk 2]
```diff
@@ -1103,7 +1070,7 @@ exports.defineAutoTests = function () {
 
                     // windows store and ios are too fast, win is called before we have a chance to abort
                     // so let's get them busy - while not providing an extra load to the slow Android emulators
-                    var arrayLength = ((isWindows && !isWindowsPhone81) || isIos) ? 3000000 : isIot ? 150000 : 200000;
```
Comment (erisu) [inline tests/tests.js:1106]:
```
var arrayLength = ((isWindows && !isWindowsPhone81) || isIos) ? 3000000 : isIot ? 150000 : 200000;
```
would become this, IF the above remains
```
var arrayLength = ((isWindows && !isWindowsPhone) || isIos) ? 3000000 : isIot ? 150000 : 200000;
```

**Review by erisu — COMMENTED** [review]
Please confirm the comments...

I am not super familiar with the existing supported windows platform but I think they can still be installed on phones.

If that is correct, then I think some of the follow needs to stay but changed slightly.

File: tests/tests.js (hunk 1 above)
Comment (timbru31) [reply tests/tests.js:58]: Good catch(es). I'll take a closer look.

File: tests/tests.js (hunk 2 above)
Comment (timbru31) [reply tests/tests.js:1106]: done

File: tests/tests.js (hunk 1 above)
Comment (timbru31) [reply tests/tests.js]: done

**Review by erisu — APPROVED** [review]

**Pull request merged**